### PR TITLE
Add blend func sprite

### PR DESCRIPTION
**Pull request by dabingnn**
> Makes sure these boxes are checked before submitting your PR - thank you!
> - [X] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
> - [X] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
> - [X] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
> - [X] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.

Re: cocos-creator/fireball#

Changes proposed in this pull request:
- add Blend function for sprite (https://github.com/cocos-creator/fireball/issues/1874)

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/components/CCSprite.js [hunk 1]
```diff
@@ -202,6 +208,40 @@ var Sprite = cc.Class({
             animatable: false
         },
 
+        /**
+         * specify the source Blend Factor
+         * @property srcBlendFactor
+         * @type {BlendFactor}
+         */
+        srcBlendFactor: {
```
Comment (jareguo) [inline cocos2d/core/components/CCSprite.js:216]: 既然是 setBlendFunc, 为什么这里不是 srcBlendFunc 而是 srcBlendFactor

File: cocos2d/core/value-types/CCTypes.js [hunk 2]
```diff
@@ -60,19 +60,36 @@ cc.BlendFunc = function (src1, dst1) {
     this.dst = dst1;
 };
 
+var BlendFactor = cc.Enum({
+    ONE: 1, //cc.ONE
+    ZERO: 0, //cc.ZERO
+    SRC_ALPHA: 0x0302, //cc.SRC_ALPHA
```
Comment (jareguo) [inline cocos2d/core/value-types/CCTypes.js]: 0x302 吧，能和下面对齐

File: cocos2d/core/components/CCSprite.js (hunk 1 above)
Comment (dabingnn) [inline cocos2d/core/components/CCSprite.js:216]: BlendFunc ＝ srcBlendFactor \* src + dstBlendFactor \* dst;
整个加起来才是blendFunction

File: cocos2d/core/value-types/CCTypes.js (hunk 2 above)
Comment (dabingnn) [inline cocos2d/core/value-types/CCTypes.js]: 多谢提醒。已改正

**Pull request merged**